File: frontend/src/api/glucose.rs
use crate::api::{get_json, post_empty, ApiResult};
use crate::models::glucose::{GlucoseReading, SyncResult};

pub async fn get_history(
    subject_id: &str,
    _hours_back: u32,
    token: &str,
) -> ApiResult<Vec<GlucoseReading>> {
    let url = format!("/api/glucose/subjects/{}/readings", subject_id);
    get_json(&url, Some(token)).await
}

pub async fn sync_history(subject_id: &str, token: &str) -> ApiResult<SyncResult> {
    let url = format!("/api/glucose/subjects/{}/sync", subject_id);
    post_empty(&url, Some(token)).await
}

File: frontend-react/src/api/glucose.ts
import client from './client';
import type { GlucoseReading, SyncResult } from '../types';

export async function getHistory(subjectId: string): Promise<GlucoseReading[]> {
  const { data } = await client.get<GlucoseReading[]>(`/api/glucose/subjects/${subjectId}/readings`);
  return data;
}

export async function syncHistory(subjectId: string): Promise<SyncResult> {
  const { data } = await client.post<SyncResult>(`/api/glucose/subjects/${subjectId}/sync`);
  return data;
}

export async function getLatest(subjectId: string | number): Promise<GlucoseReading | null> {
  try {
    const { data } = await client.get<GlucoseReading>(`/api/glucose/subjects/${subjectId}/readings/latest`);
    return data;
  } catch {
    return null;
  }
}
